astromon/cross_match.py

```diff
@@ -468,26 +699,87 @@ def _simple_cross_match(
         keys=['obsid', 'x_id'],
         table_names=['xray', 'cat']
     )
-    assert np.all(matches['id_xray'] == matches['x_id'])
-    matches.rename_column('id_cat', 'c_id')
 
-    matches['dz'] = matches['z_angle_xray'] - matches['z_angle_cat']
-    matches['dy'] = matches['y_angle_xray'] - matches['y_angle_cat']
+    matches['dz'] = matches['x_z_angle'] - matches['c_z_angle']
+    matches['dy'] = matches['x_y_angle'] - matches['c_y_angle']
     matches['dr'] = np.sqrt(matches['dy']**2 + matches['dz']**2)
     matches['cat_order'] = 200
     for i, k in enumerate(catalogs):
-        matches[matches['catalog'] == k]['cat_order'] = i
+        matches['cat_order'][matches['catalog'] == k] = i
 
-    matches = matches[matches['r_angle'] < r_angle]
+    matches = matches[
+        ((matches['grating'] == 'NONE') & (matches['r_angle'] < r_angle))
+        | ((matches['grating'] != 'NONE') & (matches['r_angle'] < r_angle_grating))
+    ]
+    matches = matches[matches['dr'] < dr]
 
     if len(matches) == 0:
         return []
 
-    matches = matches.group_by('obsid')
```

Comment:
This was the bug that was causing your version to return so many fewer sources. It was just returning one match per obsid.

Reply:
yup, that's a bug.

Still, you got more than 10 times the number of matches, and I don't think this accounts for that difference.

Reply:
After applying  ff34529 I get exactly the same number of matches (68197) in an apples-to-apples comparison. In my original code `xcorr` is every rough match for the mission (no filtering at all).
```
xcorr = cross_match.compute_cross_matches(
    catalogs=['ICRS', 'Tycho2', '2MASS', 'SDSS', 'USNO-B1.0'],
     dr=10000,
    snr=-10000,
    r_angle=10000,
    r_angle_grating=10000,
    near_neighbor_dist=-10000,
)
```